Demos/Device/ClassDriver/CCID/CCID.c

```diff
@@ -0,0 +1,223 @@
+{
+	/* Indicate USB not ready */
+	LEDs_SetAllLEDs(LEDMASK_USB_NOTREADY);
+}
+
+/** Event handler for the library USB Configuration Changed event. */
+void EVENT_USB_Device_ConfigurationChanged(void)
+{
+	bool ConfigSuccess = true;
+
+	ConfigSuccess &= CCID_Device_ConfigureEndpoints(&CCID_Interface);
+
+	/* Indicate endpoint configuration success or failure */
+	LEDs_SetAllLEDs(ConfigSuccess ? LEDMASK_USB_READY : LEDMASK_USB_ERROR);
+}
+
+/** Event handler for the library USB Control Request reception event. */
+void EVENT_USB_Device_ControlRequest(void)
+{
+	CCID_Device_ProcessControlRequest(&CCID_Interface);
+}
+
+/** Event handler for the CCID_PC_to_RDR_IccPowerOn message. This message is sent to the device
+ *  whenever an application at the host wants to send a power off signal to a slot.
+ *  THe slot must reply back with a recognizable ATR (answer to reset) 
+ */
+uint8_t CALLBACK_CCID_IccPowerOn(uint8_t slot, uint8_t * atr, uint8_t * atrSize, uint8_t *error) {
+	if(slot < CCID_Interface.Config.TotalSlots) {
+		Iso7816_CreateSimpleAtr(atr, atrSize);
+		*error = CCID_ERROR_NO_ERROR;
+		return CCID_COMMANDSTATUS_PROCESSEDWITHOUTERROR | CCID_ICCSTATUS_PRESENTANDACTIVE;
+	} else {
+		*error = CCID_ERROR_SLOT_NOT_FOUND;
+        return CCID_COMMANDSTATUS_FAILED | CCID_ICCSTATUS_NOICCPRESENT;
+	}
+}
+
+/** Event handler for the CCID_PC_to_RDR_IccPowerOff message. This message is sent to the device
+ *  whenever an application at the host wants to send a power off signal to a slot.
+ */
+uint8_t CALLBACK_CCID_IccPowerOff(uint8_t slot, uint8_t *error) {
+	if(slot < CCID_Interface.Config.TotalSlots) {
+		*error = CCID_ERROR_NO_ERROR;
+		return CCID_COMMANDSTATUS_PROCESSEDWITHOUTERROR | CCID_ICCSTATUS_NOICCPRESENT;
+	} else {
+		*error = CCID_ERROR_SLOT_NOT_FOUND;
+        return CCID_COMMANDSTATUS_FAILED | CCID_ICCSTATUS_NOICCPRESENT;
+	}
+}
+
+/** Event handler for the CCID_PC_to_RDR_GetSlotStatus. THis message is sent to the device
+ *  whenever an application at the host wants to the get the current slot status
+ *
+ */
+uint8_t CALLBACK_CCID_GetSlotStatus(uint8_t slot, uint8_t *error) {
+	if(slot < CCID_Interface.Config.TotalSlots) {
+		*error = CCID_ERROR_NO_ERROR;
+		return CCID_COMMANDSTATUS_PROCESSEDWITHOUTERROR | CCID_ICCSTATUS_PRESENTANDACTIVE;
+	} else {
+		 *error = CCID_ERROR_SLOT_NOT_FOUND;
+         return CCID_COMMANDSTATUS_FAILED | CCID_ICCSTATUS_NOICCPRESENT;
+	}
+}
+
+/** Event handler for the CCID_PC_to_RDR_XfrBlock. THis message is sent to the device
+ *  whenever an application at the host wants to send a block of bytes to the device
+ *  THe device reply back with an array of bytes
+ */
+uint8_t CALLBACK_CCID_XfrBlock(uint8_t slot, uint8_t *error, uint8_t * receivedBuffer, uint8_t receivedBufferSize, uint8_t * sendBuffer, uint8_t *sentBufferSize) {
+	if(slot < CCID_Interface.Config.TotalSlots) {
+		uint8_t okResponse[2] = {0x90, 0x00};
+		memcpy(sendBuffer, okResponse, sizeof(okResponse) * sizeof(uint8_t));
```

Comment:
`sizeof(okResponse) * sizeof(uint8_t)` should be just `sizeof(okResponse)`, as `sizeof()` evaluated on a local array (not a pointer to an array) will include the stride length in the size calculation.